app/helpers/browse_helper.rb

```diff
@@ -72,6 +72,63 @@ def link_follow(object)
     "nofollow" if object.tags.empty?
   end
 
+  # Tag change highlighting methods for history view
+  def tag_changes_for_version(current_version, all_versions)
+    return {} unless current_version && all_versions
+
+    current_tags = current_version.tags || {}
+
+    # Find the previous version by sorting all versions and finding the one before current
+    sorted_versions = all_versions.sort_by(&:version)
+    current_index = sorted_versions.find_index { |v| v.version == current_version.version }
+    previous_version = current_index&.positive? ? sorted_versions[current_index - 1] : nil
+
+    # Don't compare with redacted versions unless we're showing redactions
+    previous_tags = if previous_version && (!previous_version.redacted? || params[:show_redactions])
```

Comment:
Would it make sense to filter out the redacted versions when finding the index above, and show changes bridging versions, pretending that the redacted ones don't exist?

So for example: if we have v1, v2(redacted), v3, then show changes between v1 and v3